### PR TITLE
Ontology for knowledge graph about research papers

### PR DESCRIPTION
**Pull request by tsiudalski**
Adding an ontology describing properties used in knowledge graph about research papers from the domain of Computer Science.

### CODE REVIEW

File: schema/ont_papers.ttl [hunk 1]
```diff
@@ -0,0 +1,68 @@
+@prefix : <http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/> .
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:1]: Wrong empty prefix.

File: schema/ont_papers.ttl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xml: <http://www.w3.org/XML/1998/namespace> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix fabio: <http://purl.org/spar/fabio/> .
+@base <http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/> .
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:8]: Here as well.

File: schema/ont_papers.ttl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+<https://w3id.org/ocs/ont/papers/> rdf:type owl:Ontology ;
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:10]: Ontology IRI should not end with a / or # – only the prefix should. So, make it just `https://w3id.org/ocs/ont/papers`.

File: schema/ont_papers.ttl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                                    rdfs:comment "This ontology describes properties used in OpenCS knowledge graph about research articles from the domain of Computer Science." .
+
+#################################################################
+#    Object Properties
+#################################################################
+
+###  http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/hasRelatedPapers
+:hasRelatedPapers rdf:type owl:ObjectProperty ,
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:18]: Maybe "hasRelatedPaper"? The range of this prop is a paper, not a set of papers.

The same applies to other properties below.

File: schema/ont_papers.ttl [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                           owl:SymmetricProperty ,
+                           owl:ReflexiveProperty ;
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:20]: Are you sure that paper X is related to paper X?

File: schema/ont_papers.ttl [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+:relationScore rdf:type owl:DatatypeProperty ,
+                        owl:FunctionalProperty ;
+               rdfs:range xsd:float ;
+               rdfs:comment "Quantifies the degree of relationship between a given research paper and a topic. The higher the value, the stronger the association, with no fixed range specified." .
+
+
+###  http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/similarityScore
+:similarityScore rdf:type owl:DatatypeProperty ,
+                          owl:FunctionalProperty ;
+                 rdfs:domain fabio:ResearchPaper ;
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:55]: Something is wrong here. If I remember correctly, you attached this property not to the paper itself, but to some helper instance, one instance per relation. I think you need something like `RelatedPaperRelation` class to represent this.

Note that you don't have to specify this class explicitly in the KG – you can just model it in the ontology and we will know it's there.

Comment (Ostrzyciel) [reply schema/ont_papers.ttl:55]: Oh, fixing this will require more changes to the ontology elsewhere. :)

File: schema/ont_papers.ttl [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+###  http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/relationScore
+:relationScore rdf:type owl:DatatypeProperty ,
+                        owl:FunctionalProperty ;
+               rdfs:range xsd:float ;
+               rdfs:comment "Quantifies the degree of relationship between a given research paper and a topic. The higher the value, the stronger the association, with no fixed range specified." .
+
+
+###  http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/similarityScore
+:similarityScore rdf:type owl:DatatypeProperty ,
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:53]: This prop is named `similarityScore`, while the object props all start with `has`. Please pick one convention and stick to it.

File: schema/ont_papers.ttl [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                           owl:ReflexiveProperty ;
+                  rdfs:domain fabio:ResearchPaper ;
+                  rdfs:range fabio:ResearchPaper ;
+                  rdfs:comment "Indicates that a research paper has similar content to another research paper, determined by the cosine similarity of their embeddings." .
+
+
+###  http://www.semanticweb.org/tomas/ontologies/2023/10/untitled-ontology-3/hasRelatedTopics
+:hasRelatedTopics rdf:type owl:ObjectProperty ,
+                           owl:IrreflexiveProperty ;
+                  rdfs:domain fabio:ResearchPaper ;
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:29]: Missing range?

**Review by Ostrzyciel — COMMENTED** [review]
Some work left to do :) 

You will need to rethink the class structure a bit – please have a look especially at the comment at line 55.

File: schema/ont_papers.ttl (hunk 6 above)
Comment (tsiudalski) [reply schema/ont_papers.ttl:55]: I think there is a similar problem with the related topics relation, I am attaching a possible solution for both, is it ok?
relations

File: schema/ont_papers.ttl (hunk 1 above)
Comment (tsiudalski) [reply schema/ont_papers.ttl:1]: Sorry it was added automatically and I overlooked it, I will fix it

File: schema/ont_papers.ttl (hunk 6 above)
Comment (Ostrzyciel) [reply schema/ont_papers.ttl:55]: Looks better :)

Sugestions:
- Make a superclass (let's say `:PaperRelation`) for `:RelatedPaperRelation` and `:RelatedTopicRelation`. This way you could define the range of :hasRelation more easily (as just :PaperRelation)
- Merge the two props for scores into one, they do the same thing, right?
- For relation target – leave the range empty for now, I will show you on Friday how to make an axiom to express the range in a situation like this. For us to be able to discuss it, please send me a draft including at least these changes today, ok?

**Comment by Ostrzyciel** [comment]
@tsiudalski please mark resolved issues as "resolved" – this helps to organize the work :)

**Review by Ostrzyciel — CHANGES_REQUESTED** [review]

File: schema/ont_papers.ttl [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+#    Object Properties
+#################################################################
+
+###  https://w3id.org/ocs/ont/papers/hasRelated
+ocs_papers:hasRelated rdf:type owl:ObjectProperty ,
+                               owl:InverseFunctionalProperty ,
+                               owl:IrreflexiveProperty ;
+                      rdfs:domain fabio:ResearchPaper ;
+                      rdfs:range ocs_papers:RelatedObjectRelation ;
+                      rdfs:comment "Represents a relation between research paper and other object. Points to RelatedObjectRelation class. Other properties are then used to further describe the relation."@en .
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:26]: Please add `rdfs:label` for each class and property (for example: "Has related" for this prop). Please remember to add the labels with the "en" language tag – you can specify it in the bottom-right of the popup window in Protege when editing the value of some property.

**Review by Ostrzyciel — CHANGES_REQUESTED** [review]

File: schema/ont_papers.ttl [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xml: <http://www.w3.org/XML/1998/namespace> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix fabio: <http://purl.org/spar/fabio/> .
+@prefix ocs_papers: <https://w3id.org/ocs/ont/papers/> .
+@base <https://w3id.org/ocs/ont/papers> .
+
+<https://w3id.org/ocs/ont/papers> rdf:type owl:Ontology ;
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:12]: Missing label for the ontology itself :)

File: schema/ont_papers.ttl [hunk 11]
```diff
@@ -0,0 +1,114 @@
+@prefix : <https://w3id.org/ocs/ont/papers#> .
```
Comment (Ostrzyciel) [inline schema/ont_papers.ttl:1]: You seem to have overwritten my changes – I have added language tags for comments and added a few new axioms. Please fix this – go to your git history and find my changes. Next time, please make sure you pull before making more changes – there may be more than one person working on your branch.

**Review by Ostrzyciel — CHANGES_REQUESTED** [review]

**Pull request merged**